RUFAS/input_manager.py

```diff
@@ -53,120 +76,286 @@ def _load_metadata(self, metadata_path: str = "input/example_metadata.json") ->
-        # where element is found to be invalid, place this warning:
-        # om.add_warning("Invalid data", f"Invalid data found: {key=}; {value=}", info_map)
-        pass
+        info_map = {"class": self.__class__.__name__,
+                    "function": self._validate_element.__name__,
+                    }
+        variable_properties = reduce(lambda d, key: d[key], element_hierarchy,
+                                     self.__metadata["properties"][property_map_key])
+        var_type = variable_properties["type"]
+        is_nested = var_type == "object"
+        if is_nested:
+            children_status: Dict[str, bool] = {}
+            false_counter = 0
+            for nested_key in variable_properties.keys():
+                element_hierarchy.append(nested_key)
+                child_status = self._validate_element(self, module_key, element_hierarchy,
+                                                      property_map_key, input_data, eager_termination)
+                if eager_termination and not child_status:
+                    return False
+                element_path = ".".join(element_hierarchy)
+                children_status[element_path] = child_status
+                if not child_status:
+                    om.add_warning("Invalid nested element found", f"{element_path}", info_map)
+                    false_counter += 1
+                element_hierarchy.remove(nested_key)
+            is_valid = false_counter == 0
+            if is_valid:
+                return True
+            else:
+                invalid_child_vars = [key for key, value in children_status.items() if value is False]
+                om.add_warning("Invalid nested element(s) found",
+                               f"{invalid_child_vars=}",
+                               info_map)
+                return False
+        else:
+            var_name = element_hierarchy[-1]
+            try:
+                input_data_value = reduce(lambda d, key: d[key], element_hierarchy, input_data)
+            except KeyError as e:
+                raise KeyError(f"Key {var_name} not found in pool: {e}")
+
+            type_validation_dict = {"string":
+                                    self._validate_string_type_element(variable_properties, var_name, input_data_value),
+                                    "number":
+                                    self._validate_num_type_element(variable_properties, var_name, input_data_value),
+                                    "array":
+                                    self._validate_array_type_element(variable_properties, var_name, input_data_value),
+                                    "bool":
+                                    True}
+            is_valid = type_validation_dict.get(var_type)
+
+            if is_valid:
+                return True
+            elif is_valid is None:
+                raise Exception("Element must be type number, array, string, or bool")
```

Comment:
```suggestion
                raise Exception(f"Invalid type {var_type}: Element must be type number, array, string, or bool")
```
I think more information about the exception would be useful.

Reply:
Agreed.